src/axolotl/utils/config/models/input/v0_4_1/__init__.py

```diff
@@ -0,0 +1,931 @@
+    @classmethod
+    def fix_sharegpt_datasets(cls, datasets):
+        for idx, ds_cfg in enumerate(datasets):
+            if not ds_cfg["type"]:
+                continue
+            if ds_cfg["type"] == "sharegpt:chat":
+                LOG.warning(
+                    PendingDeprecationWarning(
+                        "`type: sharegpt:chat` will soon be deprecated. simply use `type: sharegpt` instead."
+                    )
+                )
+                datasets[idx]["type"] = "sharegpt"
+            if "sharegpt_simple" in ds_cfg["type"]:
+                LOG.warning(
+                    PendingDeprecationWarning(
+                        "`type: sharegpt_simple` will soon be deprecated. simply use `type: sharegpt` instead."
+                    )
+                )
+                datasets[idx]["type"] = datasets[idx]["type"].replace(
+                    "sharegpt_simple", "sharegpt"
+                )
+        return datasets
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_batch_size_fields(cls, root):
+        non_empty_count = sum(
+            1
+            for field in (
+                "micro_batch_size",
+                "gradient_accumulation_steps",
+                "batch_size",
+            )
+            if root.get(field)
+        )
+        if non_empty_count < 2:
+            raise ValueError(
+                "At least two of [micro_batch_size, gradient_accumulation_steps, batch_size] must be set"
+            )
+        return root
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_pretraining_w_max_steps(cls, data):
+        if data.get("pretraining_dataset") and not data.get("max_steps"):
+            raise ValueError(
+                "max_steps must be set when using iterable pretraining_dataset, Trainer can't infer length and schedule optimizer/learning rate without it!"
+            )
+        return data
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_pretraining_w_group_by_length(cls, data):
+        if data.get("pretraining_dataset") and data.get("group_by_length"):
+            LOG.warning(
+                "You probably want to disable group_by_length as it will force a streamed dataset to download completely."
+            )
+        return data
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_gptq_w_revision(cls, data):
+        if data.get("gptq") and data.get("model_revision"):
+            raise ValueError(
+                "model_revision is not supported for GPTQ models. "
+                + "Please download the model from HuggingFace Hub manually for correct branch, "
+                + "point to its path, and remove model_revision from the config."
+            )
+        return data
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_sample_packing_w_xformers(cls, root):
+        if root.get("sample_packing") and root.get("xformers_attention"):
```

Comment:
Can we have a few methods check for when sample_packing is on but not actually active due to unsupported model type?

Pseudo code
```
    def check_sample_packing_active(cls, root):
         if root.get("sample_packing") and not any(llama/mistral/qwen):
             raise ValueError(
                 "sample_packing not compatible with current model type"
             )

Reply:
I don't know that we can reliably detect the model at this step of the validation to raise a ValueError